### PR TITLE
 [tests-only] adding test for updating expiration date of a share

### PR DESCRIPTION
**Pull request by ghost**
## Description
Scenarios added :
```cucumber
Scenario: user updates expiration date of a share
```

## Related Issue

- Part of: https://github.com/owncloud/ocis/issues/8111

## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/apiSharingNg/sharedByMe.feature [hunk 1]
```diff
@@ -1226,3 +1226,75 @@ Feature: resources shared by user
...
+      | resource        | testfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    When user "Alice" updates the permissions of last share using the Graph API:
+      | resourceType       | file                 |
+      | space              | Personal             |
+      | resource           | testfile.txt         |
+      | expirationDateTime | 2200-07-15T14:00:00Z |
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/sharedByMe.feature:1243]:
```suggestion
    When user "Alice" updates the last share with the following using the Graph API:
      | resourceType       | file                 |
      | space              | Personal             |
      | resource           | testfile.txt         |
      | expirationDateTime | 2200-07-15T14:00:00Z |
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 2]
```diff
@@ -240,6 +240,20 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
...
+	 *
+	 * @param string $user
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function userUpdatesThePermissions($user, TableNode $table) {
+		$response = $this->featureContext->shareNgGetLastCreatedUserGroupShare();
+		$shareID = json_decode($response->getBody()->getContents())->value[0]->id;
+		$this->updateLinkShare($user, $table, $shareID);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:254]: `updateLinkShare` is not suitable here
```suggestion
		$this->updateShare($user, $table, $shareID);
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 3]
```diff
@@ -296,6 +310,17 @@ public function userHasCreatedTheFollowingLinkShare(string $user, TableNode  $bo
...
+	}
+
+	/**
+	 * @param string $user
+	 * @param TableNode $body
+	 * @param string $shareID
+	 *
+	 * @return void
+	 */
+	public function updateLinkShare(string $user, TableNode  $body, string $shareID) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:323]:
```suggestion
	public function updateShare(string $user, TableNode  $body, string $shareID) {
```

File: tests/acceptance/features/apiSharingNg/sharedByMe.feature [hunk 4]
```diff
@@ -1226,3 +1226,75 @@ Feature: resources shared by user
...
+  Scenario: user updates expiration date of a share
+    Given user "Alice" has uploaded file with content "hello world" to "testfile.txt"
+    And user "Alice" has sent the following share invitation:
+      | resource        | testfile.txt |
+      | space           | Personal     |
+      | sharee          | Brian        |
+      | shareType       | user         |
+      | permissionsRole | Viewer       |
+    When user "Alice" updates the permissions of last share using the Graph API:
+      | resourceType       | file                 |
```
Comment (grgprarup) [inline tests/acceptance/features/apiSharingNg/sharedByMe.feature:1240]:
```suggestion
```
Let's remove that and make the necessary changes in logic below
```
if ($resourceType === 'folder') {
			$itemId = $this->spacesContext->getResourceId($user, $space, $resource);
		} else {
			$itemId = $this->spacesContext->getFileId($user, $space, $resource);
		}
```
to
```
$itemId = $this->spacesContext->getResourceId($user, $space, $resource);
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 5]
```diff
@@ -231,6 +231,20 @@ public function userSendsTheFollowingShareInvitationUsingTheGraphApi(string $use
...
 
+	/**
+	 * @When user :user updates the last share with the following using the Graph API:
+	 *
+	 * @param string $user
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function userUpdatesThePermissions($user, TableNode $table) {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:242]:
```suggestion
	public function userUpdatesTheLastShareWithFollowingUsingGraphApi($user, TableNode $table) {
```

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 6]
```diff
@@ -327,7 +347,7 @@ public function userUpdatesLastPublicLinkShareUsingTheGraphApiWith(string $user,
 			$spaceId,
 			$itemId,
 			\json_encode($body),
-			$this->featureContext->shareNgGetLastCreatedLinkShareID()
+			$permissionID
 		);
 		$this->featureContext->setResponse($response);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:352]: setResponse in `When` step method only
```suggestion
		return $response;
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Pull request merged**